bundles/core/org.eclipse.smarthome.core.persistence/src/main/java/org/eclipse/smarthome/core/persistence/rest/PersistenceResource.java

```diff
@@ -0,0 +1,289 @@
+import org.eclipse.smarthome.core.persistence.FilterCriteria.Ordering;
+import org.eclipse.smarthome.core.persistence.HistoricItem;
+import org.eclipse.smarthome.core.persistence.PersistenceService;
+import org.eclipse.smarthome.core.persistence.QueryablePersistenceService;
+import org.eclipse.smarthome.core.types.State;
+import org.eclipse.smarthome.io.rest.RESTResource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+
+/**
+ * This class acts as a REST resource for history data and provides different methods to interact with the persistence
+ * store
+ *
+ * @author Chris Jackson - Initial Contribution
+ */
+@Path(PersistenceResource.PATH)
+@Api(value = PersistenceResource.PATH)
+public class PersistenceResource implements RESTResource {
+
+    private static final Logger logger = LoggerFactory.getLogger(PersistenceResource.class);
+
+    // The URI path to this resource
+    public static final String PATH = "persistence";
+
+    static private Map<String, PersistenceService> persistenceServices = new HashMap<String, PersistenceService>();
+
+    public void addPersistenceService(PersistenceService service) {
+        persistenceServices.put(service.getName(), service);
+    }
+
+    public void removePersistenceService(PersistenceService service) {
+        persistenceServices.remove(service.getName());
+    }
+
+    static public Map<String, PersistenceService> getPersistenceServices() {
+        return persistenceServices;
+    }
+
+    @GET
+    @Produces({ MediaType.APPLICATION_JSON })
+    @ApiOperation(value = "Gets a list of persistence services.", response = String.class, responseContainer = "List")
+    @ApiResponses(value = @ApiResponse(code = 200, message = "OK") )
+    public Response httpGetPersistenceServices(@Context HttpHeaders headers) {
+        Object responseObject = getPersistenceServiceList();
+        return Response.ok(responseObject).build();
+    }
+
+    @GET
+    @Path("/{servicename: [a-zA-Z_0-9]*}/{itemname: [a-zA-Z_0-9]*}")
+    @Produces({ MediaType.APPLICATION_JSON })
+    @ApiOperation(value = "Gets item persistence data from the persistence service.", response = ItemHistoryBean.class)
+    @ApiResponses(value = @ApiResponse(code = 200, message = "OK") )
+    public Response httpGetPersistenceItemData(@Context HttpHeaders headers,
+            @PathParam("servicename") String serviceName, @PathParam("itemname") String itemName,
+            @QueryParam("starttime") String startTime, @QueryParam("endtime") String endTime,
+            @QueryParam("page") long pageNumber, @QueryParam("pagelength") long pageLength) {
+
+        final Object responseObject = getItemHistoryBean(serviceName, itemName, startTime, endTime);
+        return Response.ok(responseObject).build();
+    }
+
+    private Date convertTime(String sTime) {
```

Comment:
Why not using the DateTimeType, so the user of the REST API can use timezone and we can remove duplicate code:

```java
DateTimeType dateTime = new DateTimeType(sTime);
return dateTime.getCalendar().getTime();
```